app/_hub/kong-inc/openid-connect/how-to/client-authentication/_mtls.md

```diff
@@ -0,0 +1,108 @@
+* `revocation`
+  * [Session Authentication](/hub/kong-inc/openid-connect/how-to/authentication/session/)
+
+For all these endpoints and for the flows supported, the plugin uses mTLS client authentication as the authentication method when communicating with the IdP, for example, to fetch the token from the token endpoint.
+
+## Requirements for an mTLS client auth configuration
+
+### Auth server configuration
+
+The feature requires the IdP to be enabled to use mTLS and X.509 client certificate authentication. For configuring this in Keycloak, refer to the [Keycloak configuration](#prerequisites) section below. For different auth servers, refer to their documentation to configure this functionality.
+
+### Kong configuration
+
+* Set the client authentication methods. Refer to the configuration reference sections:
+	* [`client_auth`](/hub/kong-inc/openid-connect/configuration/#configclient_auth)
+	* [`token_endpoint_auth_method`](/hub/kong-inc/openid-connect/configuration/#configtoken_endpoint_auth_method)
+	* [`introspection_endpoint_auth_method`](/hub/kong-inc/openid-connect/configuration/#configintrospection_endpoint_auth_method)
+	* [`revocation_endpoint_auth_method`](/hub/kong-inc/openid-connect/configuration/#configrevocation_endpoint_auth_method)
+* Configure the client certificate ID as a value for the `tls_client_auth_cert_id` field as described in the [configuration reference](/hub/kong-inc/openid-connect/configuration/#config-tls_client_auth_cert_id).
+
+#### Certificates
+
+For mTLS connections, you need to pass a certificate and key to {{site.base_gateway}}. 
+Set this up using the `/certificates` Admin API endpoint:
+
+```bash
+http -f post :8001/certificates cert@crt.pem key@key.pem
+```
+
+#### SSL verify
+
+The configuration option [`tls_client_auth_ssl_verify`](/hub/kong-inc/openid-connect/configuration/#configtls_client_auth_ssl_verify) controls whether the server (IdP) certificate is verified.
+
+When set to `true` (the default value), ensure [trusted certificate](/gateway/latest/reference/configuration/#lua_ssl_trusted_certificate) and [verify depth](/gateway/latest/reference/configuration/#lua_ssl_verify_depth) are appropriately configured so that the IdP's server certificate is trusted by Kong and the handshake can be performed successfully.
+
+## Demo
+### Prerequisites
+
+Follow these prerequisites to set up a demo Keycloak app and a Kong service and route for testing mTLS client auth.
+
+{% include_cached /md/plugins-hub/oidc-prereqs.md %}
+
+### Set up mTLS for the OIDC plugin
+
+Using the Keycloak and {{site.base_gateway}} configuration from the [prerequisites](#prerequisites), 
+set up an instance of the OpenID Connect plugin.
+
+For the demo, we're going to set up the following:
+
+* Issuer, client ID: Settings that connect the plugin to your IdP (in this case, the sample Keycloak app).
+* `client_auth` and `tls_client_auth_cert_id`: Both of these options must use TLS.
```

Comment:
I think saying "both options must use TLS" can be confusing here: `client_auth` is explicitly set to `tls_client_auth` in the example, while `tls_client_auth_cert_id` is set to the UUID of the certificate that was configured above.

Reply:
Yep, that was exactly it. Fixed now.